File: Kattis/maeting/maeting.py
import bisect

n, m = map(int, input().split())

monday = list(map(int, input().split()))
tuesday = list(map(int, input().split()))

tuesday.sort()

for i in range(n):
    foundIndex = bisect.bisect_left(tuesday, monday[i])

    if foundIndex < len(tuesday) and tuesday[foundIndex] == monday[i]:
        print(f"{monday[i]} ", end="")

print()

File: Kattis/maeting/maeting.cpp
#include <iostream>
#include <algorithm>

int main() {
    int n, m;
    std::cin >> n >> m;
    int monday[n];
    int tuesday[m];

    for (int i = 0; i < n; i++) {
        int student;
        std::cin >> student;
        monday[i] = student;
    }

    for (int i = 0; i < m; i++) {
        int student;
        std::cin >> student;
        tuesday[i] = student;
    }

    std::sort(tuesday, tuesday + m);

    for (int i = 0; i < n; i++) {
        bool found = std::binary_search(tuesday, tuesday + n, monday[i]);
        if (found)
            std::cout << monday[i] << " ";
    }

    std::cout << "\n";

    return 0;
}
